presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -165,7 +165,8 @@ public Principal getUserPrincipal()
                         return;
                     }
                 }
-                catch (GSSException e) {
+                catch (Exception e) {
```

Comment:
Also, don't catch `Exception`, since that requires special handling for `InterruptedException`. It's also not necessary since all checked exceptions are known statically. Assuming `GSSException` is the only checked exception that can be  here, this could be `GSSException | RuntimeException`